.github/workflows/rust.yml

```diff
@@ -71,6 +71,35 @@ jobs:
           # we run tests using BitVec<u64,_> which doesn't.
           args: --all-features --target wasm32-unknown-unknown
 
+  no_std:
+    name: Check no_std build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+
+      - name: Install Rust stable toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          target: aarch64-unknown-none
+          override: true
+
+      # Ensure cargo is run on the no_std target architecture. Specifying `--target` does not work here,
+      # because it automatically activates dev-dependencies, which are not no_std compatible.
+      - name: Rust create rust-toolchain.toml
+        run: echo -e "[toolchain]\ntargets = [\"aarch64-unknown-none\"]" > rust-toolchain.toml
```

Comment:
Ok, I had a dig into why this was added and I think the following makes it unnecessary (and a couple of little tidy up bits):

Add this next to the `CustomError` type:

```rust
#[cfg(not(feature = "std"))]
impl From<MyError> for CustomError {
    fn from(value: MyError) -> Self {
        Box::new(value)
    }
}
```

(and remove the `Into<CustomError>` thing in the test; the test doesn't compile with it in any case).

In `Cargo.toml`, no default features on the first two dev deps:

```toml
[dev-dependencies]
bitvec = { version = "1.0.1", default-features = false }
scale-info = { version = "2.3.0", default-features = false, features = ["bit-vec", "derive"] }
```

This should allow dev deps to compile to no_std so you can then remove from the CI file:

```rust
- name: Rust create rust-toolchain.toml
  run: echo -e "[toolchain]\ntargets = [\"aarch64-unknown-none\"]" > rust-toolchain.toml
```

Reply:
Sure, I can do that - if you're sure you want to limit the test as well to `no_std`? That might be somewhat limiting in the future and is not actually required as a library.

Reply:
I figure that this way we can run some tests in `no-std` at least (rather than not at all), and if we introduce dev deps that only make sense in std, we can make the dep optional on the "std" feature being enabled, and hide the tests behind a `#[cfg(feature = "std")]`